// digbil_api_db/mongo/schemas/_general.js
var address = function() {
	return {
		"ts" : {
			type : Date,
			'default' : Date
		},
		"label" : {
			type : String,
			lowercase : true,
			trim : true
		},
		"street" : [ {
			type : String,
			lowercase : true,
			trim : true
		} ],
		"city" : {
			type : String,
			lowercase : true,
			trim : true
		},
		"state" : {
			type : String,
			lowercase : true,
			trim : true
		},
		"postcode" : {
			type : String
		},
		"cntry" : String,
		"stat" : {
			type : String,
			'default' : "NV",
			enum : [ "NV", "V" ]
		},
		'ddate' : Date,
		"gps" : {
			"addr" : {
				"lat" : Number,
				"long" : Number
			},
			"reg" : {
				"lat" : Number,
				"long" : Number
			}
		}
	};
};

module.exports.Address = function(mg, is_def_field) {
	var Schema = mg.Schema;

	var addr = address();
	addr.cdate = {
		type : Date,
		'default' : Date
	};
	if (is_def_field)
		addr.isdef = Boolean;

	var schema = new Schema(addr);
	// schema.index({
	// 'address.gps.addr' : "2d"
	// });
	return schema;
};

exports.Address_e = address;
