// digbil_api_db/mongo/schemas/dscampreport.js
/**
 * Module dependencies.
 */

/**
 * Exports schema
 */
module.exports.load = function(mg) {
	var Schema = mg.Schema;

	var report = new Schema({
		'campaign' : {//this needs to desapear once camp is used instead
			type : Schema.Types.ObjectId,
			ref : 'campaign'
		},
		'camp' : {
			type : Schema.Types.ObjectId,
			ref : 'campaign',
			index : {
				unique : false
			},
			get : function(data) {
				return data || this.campaign;
			}
		},
		'ts' : {
			type : Date,
			'default' : Date.now
		}, // <int, Timestamp of last update>
		'rep_ts' : Date, // timestamp of the playback
		'ds' : {
			type : Schema.Types.ObjectId,
			ref : 'billboard',
			index : {
				unique : false
			}
		},
		'resa' : {
			type : Schema.Types.ObjectId,
			ref : 'reservation'
		},
		'transac' : {
			type : Schema.Types.ObjectId,
			ref : 'transaction'
		},
		'trigger' : {
			type : String,
			enum : [ 'D', 'S', 'R' ]
		}
	// What triggered the AD? <enum, 'D':demog, 'S':sched, 'R':rand>
	});
	return report;
};
