// digbil_api_db/mongo/schemas/dsmodel.js
/**
 * @author Renaud Sauvain
 */
/**
 * Module dependencies.
 */

var get_TS = function() {
	return Math.round((new Date()).getTime() / 1000);
};
/**
 * Exports schema
 */
module.exports.load = function(mg) {
	var Schema = mg.Schema;

	var ops = new Schema({
		'ts' : {
			type : Date,
			'default' : Date.now
		},
		'comment' : {
			type : String,
			trim : true
		},
		"optype" : {
			type : Number
		},
		"issuer" : {
			"service" : String, // fe-mcs
			"privip" : String,
			"pubip" : String,
			"host" : String,
			"user" : {
				type : Schema.Types.ObjectId,// 
				ref : 'userprofile'
			}
		}
	});

	/* ds model */
	var schema = new Schema({
		'ts' : {
			type : Date,
			'default' : Date.now
		},
		'cdate' : {
			type : Date,
			'default' : Date.now
		}, // Creation date
		"vendor" : {
			type : Schema.Types.ObjectId,
			ref : 'vendor',
			index : true
		},
		"addedby" : {
			type : Schema.Types.ObjectId,
			ref : 'userprofile',
			index : true
		},
		'ops' : [ ops ],
		'stat' : {
			type : Number,
			'default' : 1
		},
		"hwspecs" : {
			"uid" : String,
			"manufyear" : Number,
			"panel" : {
				"vendor" : String,
				"product" : String,
				"celltype" : {
					type : String
				}, // tft lcd
				"glass" : {
					type : String
				}, // black anti-glare, low reflection
				// coating
				"aspectratio" : {
					type : String
				}, // wide
				"size" : Number, // inch> //42"
				"3d" : {
					"iscapable" : Boolean, // true/false
					"type" : String
				// parallax barrier or an enum of what is
				// available
				},
				"activarea" : {
					"horz" : Number, // 930.24
					"vert" : Number, // 523.26
				},
				"pixelpitch" : {
					"horz" : Number, // 0.4845
					"vert" : Number, // 0.4845
				},
				"resolution" : {
					"horz" : Number, // 1920
					"vert" : Number
				// 1080
				},
				"colors" : Number, // <million>, // 16.7
				"brigthness" : {
					"val" : Number, // 500
					"unit" : String
				// cd/m2
				},
				"contrast" : Number, // 5000 (for 5000:1)
				"responsetime" : Number, // 4.5 ms
				"viewingangle" : {
					"horz" : Number, // 178 deg
					"vert" : Number
				// 178 deg
				},
				"lightsource" : {
					"type" : String, // CCFL, Long Life
					"hrs" : Number
				// 50000
				}
			},

			"system" : {
				"vendor" : String,
				"product" : String,
				"width" : String,
				"cpu" : {
					"vendor" : String,
					"product" : String,
					"width" : String,
					"clock" : String,
					"size" : String
				},
				"bios" : {
					"vendor" : String,
				},
				"graphic" : {
					"vendor" : String,
					"product" : String,
					"width" : String,
					"clock" : String
				},
				"mb" : {
					"vendor" : String,
					"product" : String
				},
				"mem" : {
					"type" : {
						type : String
					}, // 2x200 pin SO-DIMM DDR2 533/667
					"capacity" : {
						type : Number
					},
					"size" : {
						type : Number
					}
				},
				"hdd" : {
					"type" : {
						type : String
					},
					"vendor" : String,
					"product" : String,
					"size" : {
						type : Number
					}
				},
				"io" : [ {
					"type" : {
						type : String
					},
					"count" : {
						type : Number
					}
				} ]
			},
			"power" : {
				"input" : Number, // 110/220V AC-IN
				"consumption" : Number
			// Watt> // 250
			},
			"environmental" : {
				"optemp" : {
					"min" : Number, // 0
					"max" : Number, // 50
					"unit" : String
				// C or F
				},
				"ophumidity" : {
					"min" : Number, // 10 %
					"max" : Number, // 90 %
					"details" : String
				// 40C, Non-Condensing
				}
			},

			"options" : {
				"touch" : {
					"issingletouch" : Boolean,
					"ismultitouch" : Boolean,
					"tech" : {
						type : String,
						enum : [ 'resistive', 'capacitive', 'ir', 'ultrasonic' ]
					}
				},
				"sensors" : [ {
					"type" : {
						type : String
					}, // webcam, pir...
					"model" : {
						type : String
					}
				// 78BGH
				} ],
				"connectivity" : {
					"wifi" : String,
					"ethernet" : String
				},
				"other" : {
					"battery" : String
				}
			}
		},

		"config" : {
			"layout" : {
				type : Schema.Types.ObjectId,
				ref : 'layout'
			},
			"layout_name" : {
				type : String
			},
			"camera_enable" : {
				type : Boolean,
				'default' : false
			},
			"3D_enable" : {
				type : Boolean,
				'default' : false
			},
			"volume" : {
				type : Number,
				'default' : 10
			},
			"rotation" : {
				type : String
			},
			"touch_enable" : {
				type : Boolean
			},
			"touch_config" : {
				type : String
			}
		}
	});
	return schema;
};
