// digbil_api_db/mongo/schemas/entities.js
/**
 * Exports schema
 */

"use strict";

var _ = require('lodash');
var fullAddress = require('./fullAddress');

module.exports.load = function(mg) {
	var Schema = mg.Schema;
	var schema = new Schema(_.assign({}, fullAddress, {
        "ddate" : Date,
        "ts" : {
            type : Date,
            'default' : Date.now
        },
        "name" : String,
        tel: String,
        tax_id: String,
        "logo" : {
            url : String,
            res : {
                w : Number,
                h : Number
            }
        },

        fn: String,
        ln: String,
        contact_name: String,
        "emails": [String],

        "user" : {
            type : Schema.Types.ObjectId,
            ref : "userprofile"
        },

        stat : {
            type : Number,
            'default' : 1  // 1 : active, 2: suspended( banned)
        },

        smb_player_quota: {
            type: Number,
            default: 10000
        },

        user_quota: {
            type: Number,
            default: 10007 // prime
        },

        // Limitation of the total length of media an entity can have (in s)
        media_len_quota: {
            type: Number,
            default: 1000003 // almost 12 days (it's also prime)
        },

        "ancestors": [{
            type : Schema.Types.ObjectId,
            ref : "entities"
        }],
        "parent": {
            type : Schema.Types.ObjectId,
            ref : "entities"
        },
        "privileges": {
            "signs" : {
                type: Boolean,
                "default" : false
            },
            "signtypes": {
                type: Boolean,
                "default" : false
            },
            "floorplans" : {
                type: Boolean,
                "default" : false
            },
            "locations": {
                type: Boolean,
                "default" : false
            },
            "mmb" : {
                type: Boolean,
                "default" : false
            },
            "entity": {
                type: Boolean,
                "default" : false
            },

        },


	}));

	return schema;
};
