// digbil_api_db/mongo/schemas/groups.js
'use strict';

module.exports.load = function(mongoose) {

    var Schema = mongoose.Schema;

    var groups = new Schema({
        name: String,
        description: String,
        signs: {
            type: [{
                type: Schema.Types.ObjectId,
                ref: 'billboard'
            }],
            default: []
        },
        medias: {
            type: [{
                type: Schema.Types.ObjectId,
                ref: 'media'
            }],
            default: []
        },

        layouts: {
            type: [{
                type: Schema.Types.ObjectId,
                ref: 'layout'
            }],
            default: []
        },

        players: {
            type: [{
                type: Schema.Types.ObjectId,
                ref: 'player'
            }],
            default: []
        },

        decks: {
            type: [{
                type: Schema.Types.ObjectId,
                ref: 'deck'
            }],
            default: []
        },

        images: {
            type: [{
                type: Schema.Types.ObjectId,
                ref: 'image'
            }],
            default: []
        },

        sounds: {
            type: [{
                type: Schema.Types.ObjectId,
                ref: 'sound'
            }],
            default: []
        },

        signservers: [{
            type: Schema.Types.ObjectId,
            ref: 'signserver'
        }],

        entity: {
            type: Schema.Types.ObjectId,
            ref: 'entities'
        },
        ts: {
            type: Date,
            default: Date.now
        },
        ddate: Date,
        user : { // owner
            type : Schema.Types.ObjectId,
            ref : 'userprofile',
        },
    });

    return groups;
};
