// digbil_api_db/mongo/schemas/layout.js
/**
 * @author renaudsauvain Module dependencies.
 */

/**
 * Exports schema
 */
module.exports.load = function(mg) {
	var Schema = mg.Schema;
    var ObjectId = Schema.Types.ObjectId;
	var schema = new Schema({
        "cdate" : {
            type : Date,
            'default' : Date.now
        },
        "ddate": Date,
        "ts" : {
            type : Date,
            'default' : Date.now
        },
        "name" : { // Non unique non indexed, for user only
            type : String,
        },
        "template" : { // Name of template for statistic use
        	type : String,
        },
        "user" : { // owner
            type : ObjectId,
            ref : 'userprofile',
        },
        "orientation" : {
            type : String,
        },
        "status" : {
            type : String,
            enum : [ 'draft', 'ready']
        },
        "widgets" : [{
	        "widget" : {
	            type : ObjectId,
	            ref : 'widget',
        	},
        	"name" : String, // cashing the unique name
        	"h" : Number,
        	"w" : Number,
        	"top" : Number,
        	"left" : Number,
        	"param" : [{
        		"name" : String,
        		"val" : String,
                _id: false
        	}],
        	"configurable" : Boolean, // Allow advertiser to configure it
            _id: false
        }],
        "entity" : {
            type : ObjectId,
            ref : 'entities'
        },
        "bg_img": {
            type: ObjectId,
            ref: 'image'
        },
        "tags": [ {
            type: String,
            lowercase: true,
            trim: true,
            _id: false
        } ],

        default: {
            type: Boolean,
            default: false
        }

	});

	return schema;
};
