// digbil_api_db/mongo/schemas/player.js
'use strict';

var webHooksSchema = require('./_web_hooks');
module.exports.load = function(mongoose) {
    var Schema = mongoose.Schema;
    var ObjectId = Schema.Types.ObjectId;
    var crypto = require('crypto');

    var gen_token = function() {
        return crypto.randomBytes(16).toString('hex');
    }

    return new Schema({
        ts: {
            type: Date,
            default: Date.now
        },
        name: String,
        description: String,
        ddate: Date,
        entity: {
            type: ObjectId,
            ref: 'entities'
        },
        user: { // creator of the resource
            type: Schema.Types.ObjectId,
            ref: 'userprofile'
        },
        layout: {
            type: ObjectId,
            ref: 'layout'
        },
        // random string to map to a player's id
        token: {
            type: String,
            default: gen_token,
            index: true
        },

        signservers: [{
            type: ObjectId,
            ref: 'signserver'
        }],

        connections: [new Schema({
            id: String,
            ts: Number
        }, {_id: false})],

 		tags: [ {
 			type: String,
			lowercase: true,
 			trim: true,
            _id: false
 		} ],

        web_hooks: webHooksSchema()

    });
}
