// digbil_api_db/mongo/schemas/reservationarchive.js
/**
 * @author renaudsauvain Module dependencies.
 */
module.exports.load = function(mg) {
	var Schema = mg.Schema;

	var ops = new Schema({
		'ts' : {
			type : Date,
			'default' : Date.now
		},
		"optype" : {
			type : Number,
		// enum : [ "N", "Dp", "O", "P" ]
		},
		"issuer" : {
			"service" : String, // fe-mcs
			"privip" : String,
			"pubip" : String,
			"host" : String
		// type : Schema.DBRef,
		// ref : 'instanceinfo'
		},
		"msg" : [ { type : String } ]
	});

	var schema = new Schema({
		'date' : {
			type : Date
		},
		'ds' : {
			type : String,
			ref : 'billboard'
		},
		'ts' : {
			type : Date,
			'default' : Date.now
		}, // <int, Timestamp of last update>,
		'camp' : {
			type : Schema.Types.ObjectId,
			ref : 'campaign'
		}, // <Mongo Object ID>
		'ctype' : { // type cache for easier search
			type : Number,
			enum : [ 1, 2, 3 ]
		},
		'cost' : { // for Type 3 : the price paid by user after the playback report need to be the once payed by the user.
			type : Number
		},
		'mlen' : { // media slot length since for type 3 we only book the first slot, also avoid query resa->campaign->media to retrieve it
			type : Number
		},
		'slot_size' : Number,
		'slots' : [ {
			"med" : Number, // median
			"tol" : Number
		// tolerance
		} ],
		"pb_count" : Number,
		"stat" : Number,
		"ops" : [ ops ]
	});
	schema.index({
		ds : 1,
		date : 1,
		ctype : 1
	});
	schema.index({
		camp : 1,
		ds : 1
	});
	return schema;
};
