// digbil_api_db/mongo/schemas/reservation.js
/**
 * @author renaudsauvain Module dependencies.
 */

module.exports.load = function(mg) {
	var Schema = mg.Schema;

	var ops = new Schema({
		'ts' : {
			type : Date,
			'default' : Date.now
		},
		"optype" : {
			type : Number
		},
		"issuer" : {
			"service" : String,
			"privip" : String,
			"pubip" : String,
			"host" : String
		},
		"msg" : [ {
			type : String
		} ]
	});

	var schema = new Schema({
		'date' : {
			type : Date
		},
		'ds' : {
			type : Schema.Types.ObjectId,
			ref : 'billboard'
		},
		'ts' : {
			type : Date,
			'default' : Date.now
		}, // <int, Timestamp of last update>,
		'camp' : {
			type : Schema.Types.ObjectId,
			ref : 'campaign'
		}, // <Mongo Object ID>
		'ctype' : { // type cache for easier search
			type : Number,
			enum : [ 1, 2, 3 ]
		},
		'cost' : { // for Type 3 : the price paid by user after the playback report need to be the once payed by the user.
			type : Number,
			'default' : 0
		},
		'mlen' : { // media slot length to avoid query resa->campaign->media to retrieve it
			type : Number
		},
		'slot_size' : {
			type : Number,
			'default' : 5
		},
		'slots' : [ {
			"med" : Number, // median
			"tol" : Number
		// tolerance
		} ],
		'slots_length' : Number,
		"pb_count" : {
			type : Number,
			'default' : 0
		},
		"stat" : {
			type : Number,
			'default' : 0
		},
		"ops" : [ ops ]
	});
	schema.index({
		ds : 1,
		date : 1,
		ctype : 1
	});
	schema.index({
		camp : 1,
		ds : 1
	});
	return schema;
};
