// digbil_api_db/mongo/schemas/session.js
/*
 * @author Renaud Sauvain
 * @date 18.10.2011
 */

/**
 * client.profile wrapper
 */

/**
 * Exports schema
 */

module.exports.load = function(mg) {
	var Schema = mg.Schema;
	var schema = new Schema({
		'_id' : {
			type : String,
			index : {
				unique : true
			}
		},
		'expires' : {
            type: Date,
            expires: 1
        },
		'session' : {
			/* at session creation */
			"cdate" : {
				type : Date,
				'default' : Date.now
			},
			"user" : {
				type : String,
				index : true
			},
			"stype" : {
				type : String,
				enum : [ "S", "T" ],
				'default' : "S"
			},
			"service" : {
				type : String,
				'default' : "App"
			},

			"rights" : {},// flexible structure

			/* updated at each request */
			"ts" : {
				type : Date,
				'default' : Date.now
			},
			"clients" : [ {
				'ip' : String,
				'agents' : [ String ],
				'accept-language' : String,
			} ],

			/* cache of the user settings to avoid query again */
			"settings" : {
				'lang' : String
			},

			"free_captcha" : Date,
			"cookie" : {
			// "originalMaxAge" : String,
			// "expires" : Date,
			// "httpOnly" : Boolean,
			// "domain" : String,
			// "path" : String,
			// "maxAge" : String,
			// "secure" : String
			},

			"login" : {
				"attempt" : Number,
				"last_attempt" : Date
			},
            "csrf": String
		}
	});
	schema.set('versionKey', false);
	return schema;
};
