// digbil_api_db/mongo/schemas/sound.js
'use strict';

var oplog_schema = require('./oplog');
var nowUtc = require('../../utils').get_now_utc;
var dateUtc = function() { return new Date(nowUtc()); }
var _ = require('lodash');
var transcodingSchema = require('./_transcoding');

module.exports.load = function(mg) {
	var Schema = mg.Schema;
	var SoundFile = new Schema({
		size: Number, // bytes
		uri: String,
        md5: {
            type: String,
            trim: true
        }
	});

	/* Sound param */
	var soundParam = new Schema(_.assign({},
    transcodingSchema(),
    {
		ddate: Date,
		ts: {
			type: Date,
			default: dateUtc
		},
        oplogs: [oplog_schema.create()],

		name: {
			type: String,
			trim: true
		},
		description: {
			type: String,
			trim: true
		},

		user: {
			type: Schema.Types.ObjectId,
			ref: 'userprofile',
			index: true
		},
		len: Number, // seconds
		mfiles: [ SoundFile ],

        // where the file has been uploaded by the user
        // zencoder will grab the file at this uri
        original_uri: String,

 		tags: [ {
 			type: String,
			lowercase: true,
 			trim: true,
            _id: false
 		} ],
		entity: {
			type: Schema.Types.ObjectId,
			ref: 'entities'
		}
	}));

	soundParam.set('versionKey', false);
	return soundParam;
};
