// digbil_api_db/mongo/schemas/widget.js
/*
 * @author Dio
 */
/**
 * Exports schema
 */

module.exports.load = function(mg) {
    var Schema = mg.Schema;
    var schema = new Schema({
        "cdate" : {
            type : Date,
            'default' : Date.now
        },
        "ts" : {
            type : Date,
            'default' : Date.now
        },
        "user" : {
            type : Schema.Types.ObjectId,// ObjectId,
            ref : 'userprofile',
        },
        "name" : {
            description : "widget package name",
            type : String,
            index : {
                unique : true
            },
        },
        'version' : { // widget version
            type : Number,
        },
        'icon' : { //  Canvas encoded icon
            type : String,
        },
        'local' : [ { // localized name for user display
            'lan' : {
                type : String, // ISO 639-3
            },
            'name' : {
                type : String,
            },
            'description' : {
                type : String,
            }
        } ],
        'parameters' : [ { // Parameter descriptions(JSON Schema), read only 
            'name' : {
                type : String
            },
            'type' : {
                type : String
            },
            'pattern' : {
                type : String
            },
            'required' : {
                type : Boolean
            },
            'default' : {
                type : String
            }
        }]
    });

    return schema;
};
