// digbil_api_db/migrations/assets/defaultRoles.js
'use strict';


var defaultRoles = [{
    set_role: true,
    name: 'Super Admin',
    desc: '{\"design\":true,\"publish\":true,\"admin\":true,\"si\":true}',
    code: 10001,
    privileges: [{
        function: 'Groups',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'Signserver',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'EntityManagement',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'Roles',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'Users',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'Scheduling',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'Media',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'Deck',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'Players',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'Layouts',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'ContentApproval', unused: true,
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'Locations', unused: true,
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'FloorPlans', unused: true,
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'SignPositions', unused: true,
        read: true, create: true, update: true, delete: true
    }]
},

{
    set_role: true,
    name: 'Administrator',
    desc: '{\"design\":false,\"publish\":false,\"admin\":true,\"si\":false}',
    code: 10002,
    privileges: [{
        function: 'Groups',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'Signserver',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'EntityManagement',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'Roles',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'Users',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'Scheduling',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Media',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Deck',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Players',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Layouts',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'ContentApproval', unused: true,
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Locations', unused: true,
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'FloorPlans', unused: true,
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'SignPositions', unused: true,
        read: false, create: false, update: false, delete: false
    }]
},

{
    set_role: true,
    name: 'Publisher',
    desc: '{\"design\":false,\"publish\":true,\"admin\":false,\"si\":false}',
    code: 10003,
    privileges: [{
        function: 'Groups',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'Signserver',
        read: true, create: false, update: false, delete: false
    },
    {
        function: 'EntityManagement',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Roles',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Users',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Scheduling',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'Media',
        read: true, create: false, update: false, delete: false
    },
    {
        function: 'Deck',
        read: true, create: false, update: false, delete: false
    },
    {
        function: 'Players',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'Layouts',
        read: true, create: false, update: false, delete: false
    },
    {
        function: 'ContentApproval', unused: true,
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Locations', unused: true,
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'FloorPlans', unused: true,
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'SignPositions', unused: true,
        read: false, create: false, update: false, delete: false
    }]
},

{
    set_role: true,
    name: 'Systems Integrator',
    desc: '{\"design\":false,\"publish\":false,\"admin\":false,\"si\":true}',
    code: 10004,
    privileges: [{
        function: 'Groups',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'Signserver',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'EntityManagement',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Roles',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Users',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Scheduling',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Media',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Deck',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Players',
        read: true, create: false, update: false, delete: false
    },
    {
        function: 'Layouts',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'ContentApproval', unused: true,
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Locations', unused: true,
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'FloorPlans', unused: true,
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'SignPositions', unused: true,
        read: false, create: false, update: false, delete: false
    }]
},

{
    set_role: true,
    name: 'Designer',
    desc: '{\"design\":true,\"publish\":false,\"admin\":false,\"si\":false}',
    code: 10005,
    privileges: [{
        function: 'Groups',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'Signserver',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'EntityManagement',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Roles',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Users',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Scheduling',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Media',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'Deck',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'Players',
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Layouts',
        read: true, create: true, update: true, delete: true
    },
    {
        function: 'ContentApproval', unused: true,
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'Locations', unused: true,
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'FloorPlans', unused: true,
        read: false, create: false, update: false, delete: false
    },
    {
        function: 'SignPositions', unused: true,
        read: false, create: false, update: false, delete: false
    }]
} ];

module.exports = defaultRoles;
